compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -108,6 +108,19 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                     _ => span_bug!(self.cur_span(), "closure fn pointer on {:?}", src.layout.ty),
                 }
             }
+
+            DynStar => {
+                if let ty::Dynamic(data, _, ty::DynStar) = cast_ty.kind() {
+                    // Initial cast from sized to dyn trait
+                    let vtable = self.get_vtable_ptr(src.layout.ty, data.principal())?;
+                    let vtable = Scalar::from_maybe_pointer(vtable, &*self.tcx);
+                    let data = self.read_immediate(src)?.to_scalar();
```

Comment:
Might be worth adding an assertion that this is ptr-sized? You can do something like
```rust
let _assert_ptr_sized = data.to_pointer(self)?;
```

Reply:
Good idea, done!